File: sorts/bubblesort/bubblesort.java
import java.util.ArrayList;
import java.util.List;

public class bubblesort{


    public static void main(String[] args){
        Integer size = 10;
        List<Integer> theList = new ArrayList<>(new Double(size*Math.random()).intValue());

        for(Integer i = 0; i < size; i++){
            theList.add(new Double(7*size*Math.random()).intValue());
        }

        // bubble sort it
        Boolean swapped = false;
        Integer rightBuffer = 0;
        for(int i = 0; i < theList.size()-1 - rightBuffer; i++){
            if(theList.get(i) > theList.get(i+1)){
                swap(theList,i, i+1);
                swapped = true;
            }
            if(i == theList.size()-2 - rightBuffer){
                if (swapped == false){
                    break;
                }
                i = -1;
                rightBuffer++;
                swapped = false;

            }
            System.out.println(theList);

        }
        
    }

    public static void swap(List<Integer> someList,Integer a,Integer b){
        Integer temp = someList.get(a);
        someList.set(a, someList.get(b));
        someList.set(b,temp); 
    }

}

File: sorts/bubblesort/bubblesort.js


function bubbleSort(s){
    let arr = [];
    for(var i = 0; i < s; i++){
        arr.push(Math.random()*s*4);
    }
    console.log(arr);

    // way prettier with nested for loops!
    for (var right = 0; right < arr.length-1; right++){
        for(var i =0; i< arr.length-1 - right; i++ ){
            if(arr[i] > arr[i+1]){
                swap(arr,i,i+1)
            }
        }
    }

    console.log(arr);
}

function swap(arr,a,b){
    let temp = arr[a];
    arr[a] = arr[b];
    arr[b] = temp;
}

bubbleSort(10)